File: examples/python/core/invoke/invoke-with-config.py
from aws_durable_execution_sdk_python import (
    DurableContext,
    durable_execution,
)
from aws_durable_execution_sdk_python.config import Duration, InvokeConfig

@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    # Configure invoke with timeout
    invoke_config = InvokeConfig(
        timeout=Duration.from_minutes(5),
    )
    
    result = context.invoke(
        function_name="long-running-function",
        payload=event,
        name="long_running",
        config=invoke_config,
    )
    
    return result

File: examples/java/core/invoke/invoke-with-config.java
public class InvokeWithConfigHandler extends DurableHandler<Map<String, Object>, Map<String, Object>> {

    @Override
    public Map<String, Object> handleRequest(Map<String, Object> event, DurableContext ctx) {
        // Configure invoke with timeout
        InvokeConfig config = InvokeConfig.builder()
            .timeout(Duration.ofMinutes(5))
            .build();

        var result = ctx.invoke(
            "long_running",
            "long-running-function",
            event,
            Map.class,
            config
        );

        return result;
    }
}
